src/desktop/files/DesktopFileFacade.ts

```diff
@@ -138,49 +171,94 @@ export class DesktopDownloadManager {
 		}
 	}
 
-	/**
-	 * Save {@param file} to the disk. Will pick the path based on user download dir preference and the name contained
-	 * in {@param file}.
-	 */
-	async writeDataFile(file: DataFile): Promise<FileUri> {
-		const tempDir = await this.ensureUnencrytpedDir()
-		// We do not check if file exists and then write, this is generally a no-no for security reasons. We write and catch instead.
-		const files = await this.fs.promises.readdir(tempDir)
-		const filename = nonClobberingFilename(files, file.name)
-		const savePath = path.join(tempDir, filename)
-		try {
-			await this.fs.promises.writeFile(savePath, file.data)
-		} catch (e) {
-			if (e.code === "EEXIST") {
-				// if the file exists we will try again and will pick another name
-				return this.writeDataFile(file)
-			} else {
-				throw e
-			}
+	async openFileChooser(boundingRect: IpcClientRect, filter: ReadonlyArray<string> | null): Promise<Array<string>> {
+		const opts: Record<string, unknown> = { properties: ["openFile", "multiSelections"] }
```

Comment:
I know it's not new but `opts` is actually `OpenDialogOptions` right? it just wants mutable filter for no reason but we can just give it a copy anyway